codejail_service/startup_check.py

```diff
@@ -120,7 +120,14 @@ def _check_escape_subprocess():
 
     if error_message is None:
         return f"Expected error, but code ran successfully. Globals: {globals_out!r}"
-    if "Permission denied" not in error_message:
-        return f"Expected permission error, but got: {error_message}"
+    # Exact error depends on the codejail NPROC settings
+    expected_errors = [
+        # Resource-limit-based denial, e.g. NPROC=1 (can't even fork in order to exec)
+        "BlockingIOError: [Errno 11] Resource temporarily unavailable",
+        # AppArmor-based denial (confinement doesn't permit executing other binaries)
+        "PermissionError: [Errno 13] Permission denied",
+    ]
+    if not any(error in error_message for error in expected_errors):
+        return f"Expected permission or resource limit error, but got: {error_message}"
```

Comment:
Does this text change without a corresponding unit test change show a lack of coverage, or is it written without looking for this message?

Reply:
The latter -- `test_success_and_failure_modes` has coverage of this branch but doesn't look at the exact log messages. `test_logging` covers one possible failure log from each check, but not all of them. We actually don't have a lot of control over what happens in `test_logging`, though -- it has to run without confinement, so we get what we get. (We can't control which branch is taken there *and* test what happens without confinement.)